synfig-studio/plugins/lottie-exporter/layers/text.py

```diff
@@ -0,0 +1,229 @@
+"""
+Will store all the functions corresponding to text layer in lottie
+"""
+
+import sys
+import settings
+from common.Count import Count
+from helpers.transform import gen_helpers_transform
+from helpers.blendMode import get_blend
+from common.misc import is_animated
+from properties.value import gen_properties_value
+from properties.valueKeyframed import gen_value_Keyframed
+from common.misc import get_frame, parse_position
+sys.path.append("..")
+
+def calc_default_text_properties(layer):
+    """
+    Generated the lottie dictionary corresponding to properties which won't change in text-animation
+
+    Args:
+        layer  (common.Layer.Layer) : Synfig format text layer
+
+    Returns:
+        Lottie format default dictionary for text animation properties
+    """
+    default_dict = {
+      "nm": "",
+      "s": {
+        "t" : 0,
+        "xe": {},
+        "ne": {},
+        "a" : {},
+        "b" : 1,
+        "rn": 0,
+        "sh": 1,
+        "r" : 1
+      }
+    }
+    gen_properties_value(default_dict["s"]["xe"],
+                             0,
+                             7,
+                             settings.DEFAULT_ANIMATED,
+                             settings.NO_INFO)
+    gen_properties_value(default_dict["s"]["ne"],
+                             0,
+                             8,
+                             settings.DEFAULT_ANIMATED,
+                             settings.NO_INFO)
+    gen_properties_value(default_dict["s"]["a"],
+                             100,
+                             4,
+                             settings.DEFAULT_ANIMATED,
+                             settings.NO_INFO)
+
+    default_dict["nm"] = layer.get_description()
+
+    return default_dict
+def calc_anchor_alignment(lottie):
+    """
+    Generates the dictionary corresponding to the anchor alignment properties
+    
+    Args:
+        lottie (dict) : Lottie generated text layer stored here
+    """
+    lottie["t"]["m"] = {  
+                          "g": 1,
+                          "a": {}
+                       } #Has no corresponding synfig property, so using default values
+    gen_properties_value(lottie["t"]["m"]["a"],
+                             [0,0],
+                             2,
+                             settings.DEFAULT_ANIMATED,
+                             settings.NO_INFO)
+
+def calculate_text_animation(lottie,animated):
+    """
+    Generates the dictionary corresponding to text properties
+
+    Args:
+        lottie  (dict)                 : Lottie generated text properties dictionary
+        animated (lxml.etree._Element) : Synfig format animation
+    """
+    
+    waypoints = len(animated)
+    for i in range(waypoints):
```

Comment:
Could you be more informative here. What does 'i' mean here? We might be able to replace it with variable like 'waypoint'?
I know that the current code base has lots of places where we are using i and j as variables. But, I think it will be better for understanding. So despite the variable lengths, let us try to be more informative. ^__^

Reply:
Done ^__^